packages/back-end/src/services/experiments.ts

```diff
@@ -669,6 +666,207 @@ export async function createSnapshot({
   return queryRunner;
 }
 
+export type SnapshotAnalysisParams = {
+  experiment: ExperimentInterface;
+  organization: OrganizationInterface;
+  analysisSettings: ExperimentSnapshotAnalysisSettings;
+  metricMap: Map<string, ExperimentMetricInterface>;
+  snapshot: ExperimentSnapshotInterface;
+};
+
+export async function createMultipleSnapshotAnalysis(
+  params: SnapshotAnalysisParams[],
+  context: ReqContext
+): Promise<void> {
+  const analysisParamsMap = new Map<
+    string,
+    ExperimentMetricAnalysisParams & {
+      analysis: ExperimentSnapshotAnalysis;
+      unknownVariations: string[];
+      snapshotSettings: ExperimentSnapshotSettings;
+      organization: string;
+      snapshot: string;
+    }
+  >();
+  await params.forEach(
```

Comment:
Cannot await `forEach`.  The only async part is `getQueryMap`, which looks up queries by id.  Since we have all snapshots in an array, we could build an array of all queries in all snapshots and fetch them all in a single Mongo call.  Then, building the query map for each analysis will not be async

Reply:
Done